test/api/object_lifeline_test.go

```diff
@@ -222,18 +293,31 @@ func TestLifeline_receiveNewObjectStates(t *testing.T) {
 	err = heavymock.ImportRecords(ts.ConMngr.ImporterClient, lifeline.StateRecords[2].Records)
 	require.NoError(t, err)
 	// expected records from pulses 1, 2
-	ts.WaitRecordsCount(t, recordsInPulse*2, 1000)
+
+	ts.BE.PulseClient.NextFinalizedPulseFunc = func(ctx context.Context, in *exporter.GetNextFinalizedPulse, opts ...grpc.CallOption) (*exporter.FullPulse, error) {
+		p := uint32(ts.ConMngr.Importer.GetLowestUnsentPulse())
+		if p == 1<<32-1 {
+			return nil, errors.New("unready yet")
+		}
+		return clients.GetFullPulse(p), nil
+	}
+
+	ts.StartBE(t)
+	defer ts.StopBE(t)
+
+	ts.WaitRecordsCount(t, recordsInPulse*2+2, 1000)
 
 	err = heavymock.ImportRecords(ts.ConMngr.ImporterClient, lifeline.StateRecords[3].Records)
 	err = heavymock.ImportRecords(ts.ConMngr.ImporterClient, lifeline.StateRecords[4].Records)
 	require.NoError(t, err)
 	// expected records from pulses 1, 2, 3, 4
-	ts.WaitRecordsCount(t, recordsInPulse*4, 1000)
+
+	ts.WaitRecordsCount(t, recordsInPulse*4-2, 1000)
 
 	c := GetHTTPClient()
 	response, err := c.ObjectLifeline(t, lifeline.ObjID.String(), &client.ObjectLifelineOpts{Limit: optional.NewInt32(100)})
 	require.NoError(t, err)
-	require.Len(t, response.Result, recordsInPulse*4)
+	require.Len(t, response.Result, recordsInPulse*4-2)
```

Comment:
I don't understand why we expect minus 2? the logic can't be changed

Reply:
this is count of sended pulses, when we read records, we need take last nextpulse record in account

Reply:
if `2 == recordsInPulse` it's better to use variable

Reply:
It is even worst than you think, this test was completely wrong, see now.